src/ucp/tag/rndv.c

```diff
@@ -44,26 +126,77 @@ static void ucp_rndv_rma_request_send_buffer_dereg(ucp_request_t *sreq)
 
 static size_t ucp_tag_rndv_pack_send_rkey(ucp_request_t *sreq, void *rkey_buf, uint16_t *flags)
 {
-    ucp_ep_h ep = sreq->send.ep;
-    ucp_lane_index_t lane = ucp_ep_get_rndv_get_lane(ep, 0);
+    ucp_ep_h ep           = sreq->send.ep;
+    uint8_t *buf          = rkey_buf;
+    uint8_t *cnt          = buf;
+    size_t packet;
+    ucp_lane_index_t lane;
     ucs_status_t status;
+    int i;
+    ucp_rndv_rkey_data_t *rkey;
 
     ucs_assert(UCP_DT_IS_CONTIG(sreq->send.datatype));
 
-    /* Check if the sender needs to register the send buffer -
-     * is its datatype contiguous and does the receive side need it */
-    if (ucp_ep_rndv_md_flags(ep, 0) & UCT_MD_FLAG_NEED_RKEY) {
-        status = ucp_request_rndv_buffer_reg(sreq);
-        ucs_assert_always(status == UCS_OK);
+    *cnt   = 0;
+    packet = 1; /* packet at least counter */
```

Comment:
do you mean `packet_len` or `packed`? Just `packet` looks misleading

Reply:
renamed